audition/selection_rule_performance.py

```diff
@@ -0,0 +1,155 @@
+            regret_metric (string) -- model evaluation metric, such as 'precision@'
+            regret_parameter (string) -- model evaluation metric parameter,
+                    such as '300_abs'
+            model_group_ids (list of integers) The model group ids to include
+                in calculating the data
+            train_end_times (list of timestamps) The timestamps to include in
+                calculating the data
+            plot_type (string) The plot type to show (either 'regret' or 'metric')
+        """
+        df = self.generate_plot_data(
+            bound_selection_rules=bound_selection_rules,
+            model_group_ids=model_group_ids,
+            train_end_times=train_end_times,
+            regret_metric=regret_metric,
+            regret_parameter=regret_parameter,
+        )
+        if plot_type == 'regret':
+            self.regret_plot_from_dataframe(
+                metric=regret_metric,
+                parameter=regret_parameter,
+                df=df
+            )
+        elif plot_type == 'metric':
+            self.metric_plot_from_dataframe(
+                metric=regret_metric,
+                parameter=regret_parameter,
+                df=df
+            )
+        else:
+            raise ValueError('Plot type must be either regret or metric')
+
+    def generate_plot_data(
+        self,
+        bound_selection_rules,
+        model_group_ids,
+        train_end_times,
+        regret_metric,
+        regret_parameter
+    ):
+        """Create a dataframe suitable for plotting regrets over time
+
+        Args:
+            bound_selection_rules (list of .selection_rules.BoundSelectionRule)
+               Selection rule/parameter combinations to plot
+            model_group_ids (list of integers) The model group ids to include
+                in calculating the data
+            train_end_times (list of timestamps) The timestamps to include in
+                calculating the data
+            regret_metric (string) -- model evaluation metric, such as 'precision@'
+            regret_parameter (string) -- model evaluation metric parameter,
+                such as '300_abs'
+
+        Returns: (pandas.DataFrame) A dataframe with columns 'regret',
+            'train_end_time', and 'selection_rule'
+        """
+        accumulator = list()
+        for selection_rule in bound_selection_rules:
+            results = self.selection_rule_picker.results_for_rule(
+                selection_rule,
+                model_group_ids,
+                train_end_times,
+                regret_metric,
+                regret_parameter,
+            )
+            for result in results:
+                accumulator.append({
+                    'train_end_time': result['train_end_time'],
+                    'regret': result['dist_from_best_case_next_time'],
+                    'selection_rule': selection_rule.descriptive_name,
+                    'raw_value': result['raw_value'],
```

Comment:
looks like you need to include `raw_value_next_time` here -- I'm getting a key error when trying to run that plot